integration_tests/src/main/python/dpp_test.py

```diff
@@ -27,7 +27,9 @@ def fn(spark):
             ('key', IntegerGen(nullable=False, min_val=0, max_val=9, special_cases=[])),
             ('skey', IntegerGen(nullable=False, min_val=0, max_val=4, special_cases=[])),
             ('ex_key', IntegerGen(nullable=False, min_val=0, max_val=3, special_cases=[])),
-            ('value', int_gen),
+            # non-positive values here can produce a degenerative join, so we want a filter value associated
+            # with only positive values. See https://github.com/NVIDIA/spark-rapids/issues/10147
+            ('value', IntegerGen(min_val=1, special_cases=[1, INT_MAX])),
```

Comment:
Theoretically I don't think this solves the issue, although it definitely makes it better in practice.  I think there could still be a sequence of values for which we don't hit since it's random what we're generating within each range (e.g.: one side always picks the lower half of the range, the other side picks the upper half, and nothing joins).

Instead I'd recommend using RepeatSeqGen so we're _guaranteed_ to have values that match on both sides.  I'm not sure we need randomness in this test for the values -- we're not trying to test the join itself, rather just the DPP logic around the join.  If we still want to use randomness in the values, we could union the dataframe with another non-random dataframe on both sides so we are guaranteed at least some rows will match.